### PR TITLE
Post comp revision of the standard

### PR DESCRIPTION
**Pull request by amihaila**
- Add requirement for hard kill switch on some systems.
- Remove items that are no longer relevant. (Read the commit to see
  what those are >.>)
- Minor clarification & rewording.

### CODE REVIEW

File: standard.tex [hunk 1]
```diff
@@ -26,67 +26,72 @@
...
+\item Use positive fit connectors wherever possible.
 \end{itemize}
-%end of connections subsection -- }}}
+%-- }}}
 
-%start of wiring and assembly subsection -- {{{
+%Wiring and Assembly -- {{{
 \subsection{General Wiring \& Assembly}
 \begin{itemize}
+\item Any electrical system that may be unsafe to approach while powered (eg. if they contain energetics) should have a manual kill switch installed somewhere accessible.
```
Comment (jacobdeery) [inline standard.tex]: I'd elaborate here a bit. Maybe something like the following:

> Any electrical system that may be unsafe to approach while powered (e.g. if they contain energetics) should have a kill switch installed that:
> - disables all energetics or other potentially unsafe components
> - requires only a single action to operate
> - can be operated without requiring any disassembly

File: standard.tex [hunk 2]
```diff
@@ -26,67 +26,72 @@
...
+    \begin{itemize}
+    \item Wires that get handled, moved, or torqued at any point should be strain relieved on both sides of that point.
+    \item Wires should be relieved to the point where they can survive very violent shaking.
+    \item Avoid tight wires between two points.
+    \end{itemize}
+\item RC lithium polymer batteries should be fused (they can sink up to 50 A).
 \item When designing a box, circuit layout, or other assembly, keep in mind that wires and electrical components do take up space. Leave clearance in the design stage; this problem should be solved before fabrication/assembly.
-\item Keep good isolation. That means standoffs and that means decent creepage distances.
-\item When using shielded cable, ensure the shield is properly terminated and grounded. Shields should only be grounded at one end of the cable.
+\item Use standoffs to make sure the circuit board has good isolation from what's beneath it. This is especially important if the board is mounted on a metal plate (for obvious reasons).
```
Comment (jacobdeery) [inline standard.tex]: If the reasons are obvious, there's no need to say "for obvious reasons".

File: standard.tex [hunk 3]
```diff
@@ -98,54 +103,42 @@ \subsection{PCB Design}
 \end{figure}
 
 \item Don't print PCBs at the 3D Printing Center in E5. These are typically poor quality.
+\item Make sure that soldermask clearance is set such that there is soldermask between component pads (pay attention to fine pitched components) and that there is no mask on the pads themselves.
+\item Thoroughly check the Gerber files for the board before you send them out for fabrication.
 
 \item Points that only apply to designs intended to be handsoldered:
```
Comment (jacobdeery) [inline standard.tex]: "handsoldered" or "hand-soldered"? Nitpick, but we should be consistent.

File: standard.tex [hunk 4]
```diff
@@ -98,54 +103,42 @@ \subsection{PCB Design}
...
-\item Don't short things. If you have any bare wires flailing around, don't apply power to the system.
-\item All battery connections should be terminated in a female connector to prevent battery shorting and enable easy battery swapping.
-\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digikey table values, they are occasionally mislabeled.
+\item Buy spares for all components provided it is not prohibitively expensive.
+\item All battery connectors should be terminated in a female connector to prevent battery shorting and enable easy battery swapping.
+\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digikey table values, they are occasionally mislabeled. Design reviewers should also do this.
```
Comment (jacobdeery) [inline standard.tex]: DigiKey

File: standard.tex [hunk 5]
```diff
@@ -98,54 +103,42 @@ \subsection{PCB Design}
...
-\item This probably just needs to be a single checklist/procedure.
-\end{itemize}
-\item Troubleshooting Guide
-\begin{itemize}
-\item Include the symptoms of every failure mode, what the worst case scenario is, how to diagnose this failure mode conclusively, and how to fix it.
-\item This should have a list of every component in your system, and how to test that component in isolation. This list will be useful for building spares kits.
-\end{itemize}
-\item Design Documentation
-\begin{itemize}
-\item A document outlining the major decisions made for your project. The point of this is so that future generations maintaining your project know the reasoning behind your decisions. This can include meeting minutes, block diagrams, or anything else you think would be helpful. If it makes sense to reference slack conversations here, use a permalink to the relevant message.
```
Comment (jacobdeery) [inline standard.tex:148]: Why was this removed?

File: standard.tex [hunk 6]
```diff
@@ -6,10 +6,10 @@
...
 \date{\today}
-\Version{0.4}
+\Version{1.0}
 
 \begin{document}
-% start of titlepage -- {{{
+% Title page -- {{{
 \begin{titlepage}
 	\centering
 	{\scshape\Huge Waterloo Rocketry Team \par}
```
Comment (jacobdeery) [inline standard.tex]: Lose the "Team"

**Review by jacobdeery — CHANGES_REQUESTED** [review]

File: standard.tex (hunk 4 above)
Comment (amihaila) [reply standard.tex]: Digi-Key

File: standard.tex (hunk 1 above)
Comment (amihaila) [reply standard.tex]: Fixed, but I'm leaving out the "single action" requirement.

File: standard.tex (hunk 2 above)
Comment (amihaila) [reply standard.tex]: >.>

Comment (amihaila) [reply standard.tex]: \>.>

File: standard.tex (hunk 3 above)
Comment (amihaila) [reply standard.tex]: hand_soldered

File: standard.tex (hunk 5 above)
Comment (amihaila) [reply standard.tex:148]: Mostly cause none of the electrical projects this year did anything. We also have pretty high project turnover (though an argument could be made that projects that evolve from year to year should really have this). 

I'd be willing to keep it if the other people on this review are also willing, but if we're not gonna follow through I would rather remove it from the standard.

File: standard.tex (hunk 6 above)
Comment (amihaila) [reply standard.tex]: fixed

File: standard.tex (hunk 5 above)
Comment (jacobdeery) [reply standard.tex:148]: "We haven't done documentation this year" isn't good justification for not mandating documentation, in my opinion. The appropriate action here seems to be for everyone to make documentation, not to shrug and give up on trying to document our projects.

At a bare minimum I think block diagrams/architecture description should be required. We don't need to write a report for everything but we should have adequately documented what each system does and how it does it.

File: standard.tex [hunk 7]
```diff
@@ -26,67 +26,76 @@
...
 % end titlepage -- }}}
 
-%start of guiding principles section -- {{{
+%Guiding Principles --  {{{
 \section{Guiding Principles of all Decisions}
 This document exists to set out a standard for all the electrical gear (avionics, telemetry, ground support equipment). Changes/revisions are very welcome. All design decisions on mission critical electrical systems should be made with the following four points in mind:
 
 \begin{enumerate}
-\item Our electrical stuff needs to work 100\% of the time at competition. If that's unattainable, 6 sigma is also acceptable. A lot of work from a lot of people go into building this rocket, and missing our 1 launch attempt because of an electrical problem is not acceptable.
+\item Our electrical stuff needs to work 100\% of the time at competition. A lot of work from a lot of people go into building this rocket, and missing our 1 launch attempt because of an electrical problem is not acceptable.
```
Comment (akmorrison) [inline standard.tex:34]: Might be worth adding that this is true of engine tests too (to a lesser degree). Maybe institutionalize an electrical integration test at least 5 days prior to any engine test?

File: standard.tex [hunk 8]
```diff
@@ -26,67 +26,76 @@
...
-\item If you absolutely need to solder stranded core to something, make sure it's very strain relieved (I personally like P clamps, but zip ties work too).
-\item Crimped connections should be done on stranded wire only. Don't crimp onto solid core. It's weak and it will break, often when you least want it to.
-\item All exposed wire should be heatshrunk; all heatshrink should be clear (IREC regulation).
-\item If you screw up a soldering job (we all do it), redo the job. When soldering surface mount components, inspect solder joints with a small magnifying glass.
+\item Crimped connections are preferable to soldered connections. Bad crimped connections are easier to identify than bad solder joints. Additionally, it is easier to fix a bad crimp in the field since it does not require powered tooling.
+\item If you're making a crimped connection, use a proper crimp tool if at all possible. Pliers are not an acceptable alternative (see ESRA safety critical wiring guidelines). We have proper crimp tools for: generic butt splice and ring/spade connectors, Dupont connectors, and RJ45 connectors.
```
Comment (akmorrison) [inline standard.tex:50]: we have a ferrule crimper as well

File: standard.tex [hunk 9]
```diff
@@ -26,67 +26,76 @@
...
 
-%start of wiring and assembly subsection -- {{{
+%Wiring and Assembly -- {{{
 \subsection{General Wiring \& Assembly}
 \begin{itemize}
+\item Any electrical system that may be unsafe to approach while powered should have a manual kill switch installed that:
+    \begin{itemize}
+    \item disables all energetics or potentially unsafe components.
+    \item can be operated without requiring any disassembly.
+    \end{itemize}
```
Comment (akmorrison) [inline standard.tex:71]: not just operated without dissassembly - without taking rocket off of tower

File: standard.tex [hunk 10]
```diff
@@ -26,67 +26,76 @@
...
-\item If your wires get handled or moved or torqued at any point, they should be strain relieved on both sides of the torque point.
-\item Strain relieve everything to the point that it can survive violent shaking (if it's actually going in the rocket, it should survive very violent shaking).
-\item Tight wires between two points are accidents waiting to happen. Avoid these.
-\item If you're using an RC lithium polymer battery, it should be fused (those fuckers can sink 50 amps).
+\item All wires used should be rated for the current going through them.
+\item Strain relief:
+    \begin{itemize}
+    \item Wires that get handled, moved, or torqued at any point should be strain relieved on both sides of that point.
+    \item Wires should be relieved to the point where they can survive very violent shaking.
+    \item Avoid tight wires between two points.
```
Comment (akmorrison) [inline standard.tex:78]: Why? I mean, I know why, but we might want to elaborate here

File: standard.tex [hunk 11]
```diff
@@ -26,67 +26,76 @@
...
-%end of wiring and assembly subsection -- }}}
+%-- }}}
 
-%start of software subsection -- {{{
+%Software -- {{{
 \subsection{Software}
 \begin{itemize}
-\item If your system has any software anywhere, that software should be unit tested.
 \item If your system requires a computer program to be used, that program counts as mission critical. It should run on at least two machines and at least two people should be trained to use it.
-\item All mission critical software should be expressible (and accompanied by) less than 100 lines of pseudocode.
```
Comment (akmorrison) [inline standard.tex:80]: glad this is gone. Don't know why I put it there in the first place

File: standard.tex [hunk 12]
```diff
@@ -98,54 +107,42 @@ \subsection{PCB Design}
 \end{figure}
 
 \item Don't print PCBs at the 3D Printing Center in E5. These are typically poor quality.
+\item Make sure that soldermask clearance is set such that there is soldermask between component pads (pay attention to fine pitched components) and that there is no mask on the pads themselves.
+\item Thoroughly check the Gerber files for the board before you send them out for fabrication.
```
Comment (akmorrison) [inline standard.tex]: Why? I mean, I know why, but we should elaborate that the gerbers are the be all end all final product - reviews happen on the gerbers with a commit hash associated with them and they are the same regardless of machine.

Also, what do I check them for?

File: standard.tex [hunk 13]
```diff
@@ -98,54 +107,42 @@ \subsection{PCB Design}
...
 
-\item Points that only apply to designs intended to be handsoldered:
+\item Points that only apply to designs intended to be hand-soldered:
 \begin{itemize}
-\item Use SMD pads that are longer than you need. KiCad has specific footprints whose names end in \_HandSoldering, which have extended pads for easier hand-soldering (eg don't use footprint "R\_0805", use "R\_0805\_HandSoldering").
-\item SOT-23 and 0805 SMT components are quite easily hand-solderable. SOT-23-5 are also doable, but are harder.
-\item BGA components can't be hand-soldered. There might exist a reflow oven somewhere on campus that we have access to, but laying paste will be a pain. Avoid BGA packages if you plan to assemble by hand.
+\item Use SMD pads that are longer than you need. KiCad has specific footprints whose names end in \_HandSoldering, which have extended pads for easier hand-soldering (eg. use "R\_0805\_HandSoldering" instead of "R\_0805".
```
Comment (akmorrison) [inline standard.tex]: Missing close paren)

File: standard.tex [hunk 14]
```diff
@@ -98,54 +107,42 @@ \subsection{PCB Design}
...
-\item Buy spares. You will fuck things up, be prepared.
-\item Don't solder at hotter than 700\degree. Even that is probably pretty aggressive. Use leaded solder, it melts at a low temperature so you won't burn off all your flux. If your solder isn't melting below 700\degree, the iron's tip is probably bad and should be replaced. Take care of your tips!
-\item Don't short things. If you have any bare wires flailing around, don't apply power to the system.
-\item All battery connections should be terminated in a female connector to prevent battery shorting and enable easy battery swapping.
-\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digikey table values, they are occasionally mislabeled.
+\item Buy spares for all components provided it is not prohibitively expensive.
+\item All battery connectors should be terminated in a female connector to prevent battery shorting and enable easy battery swapping.
```
Comment (akmorrison) [inline standard.tex:135]: we don't design batteries. We design things that plug _into_ batteries, and those need male connectors

File: standard.tex [hunk 15]
```diff
@@ -98,54 +107,42 @@ \subsection{PCB Design}
...
+\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digi-Key table values, they are occasionally mislabeled. Design reviewers should also do this.
```
Comment (akmorrison) [inline standard.tex]: proof of this should happen before a design review - reviewee posts a document saying what they've checked

File: standard.tex [hunk 16]
```diff
@@ -98,54 +107,42 @@ \subsection{PCB Design}
...
+%-- }}}
+%-- }}}
 
-%start of documentation section -- {{{
+%Documentation -- {{{
 \section{Documentation Standards}
 It is your responsibility to ensure that your project is fully documented and available to the team. Development and documentation for all electrical projects should be tracked in git and hosted on GitHub (under the team organization, waterloo-rocketry). If you manage your repository using issues and pull requests, and if decisions regarding these are made in Slack conversations, put a permalink to the conversation in the issue comment (this is to help traceability). In general, how you manage your project repository is up to you.
 
-Release documentation for each project should also be put in Electrical/$\langle$project folder$\rangle$ on the Drive. Project documentation should include the following:
+Project documentation should include the following:
```
Comment (akmorrison) [inline standard.tex:145]: Do we say where it should live? If we don't specify "in the git repo", it might just be posted to slack (which I think is bad)

File: standard.tex (hunk 5 above)
Comment (akmorrison) [reply standard.tex:148]: We do need to write a report actually. why don't we just specify what you need for the irec report, and we'll get that done early?

File: standard.tex [hunk 17]
```diff
@@ -156,21 +153,18 @@ \section{Documentation Standards}
...
-\item Design reviews should be thorough. Even innocent looking sections should be discussed. Chances are you won't find a bug there, but it will save you pain down the line. Discuss failure modes and alternative solutions.
-\item The person doing your review should know what they're doing.
-\item If appropriate, invite junior members to listen in on the review. They will benefit from being part of the review process and will become better designers and reviewers in the future.
+\item  Conduct design reviews on every major revision of your system. This means bringing in another person to discuss your circuit. Keep a record of what was discussed and decided in the review. When presenting a new revision, indicate what has been changed since the last review (a commit message is enough).
+\item Design reviews should be thorough. Every component and subcircuit should be checked.
+\item Design reviews should be tied to a particular commit in git.
```
Comment (akmorrison) [inline standard.tex:193]: this is so you can't lie and say "oh yeah, don't worry about that, I totes fixed it in a future thing". No. We are reviewing this commit, and in this commit, this thing is broken.

Comment (akmorrison) [reply standard.tex:193]: Also, should we add "after every design review, there should be a list of things that need fixing. When you fix these things, that fix must be indicated in a commit message, so we know where problems were removed"

File: standard.tex [hunk 18]
```diff
@@ -156,21 +153,18 @@ \section{Documentation Standards}
...
 \begin{itemize}
-\item  Conduct design reviews on every major revision of your system. This means bringing in another person to discuss your circuit. Keep a record of what was discussed and decided in the review. When presenting a new revision, indicate what has been changed since the last review.
-\item Design reviews should be thorough. Even innocent looking sections should be discussed. Chances are you won't find a bug there, but it will save you pain down the line. Discuss failure modes and alternative solutions.
-\item The person doing your review should know what they're doing.
-\item If appropriate, invite junior members to listen in on the review. They will benefit from being part of the review process and will become better designers and reviewers in the future.
+\item  Conduct design reviews on every major revision of your system. This means bringing in another person to discuss your circuit. Keep a record of what was discussed and decided in the review. When presenting a new revision, indicate what has been changed since the last review (a commit message is enough).
```
Comment (akmorrison) [inline standard.tex]: Every major revision is ambiguous

I'd say it at least needs to happen:
- After you have a schematic before you start designing a PCB
- After you have a PCB mostly done (except minor things like silkscreen and such)
- Immediately before you tape out gerbers. You need a commit that has gerbers in it that has been reviewed (in gerb view, don't review gerbers from a kicad_pcb file), and passed, and then those gerbers (which are SHA verified) are what go to the fab

**Review by akmorrison — CHANGES_REQUESTED** [review]

File: standard.tex (hunk 7 above)
Comment (amihaila) [reply standard.tex:34]: I don't see that requirement as a guiding principle.

File: standard.tex (hunk 8 above)
Comment (amihaila) [reply standard.tex:50]: I left that out on purpose because our ferrule crimper is awful. It doesn't work at all for small ferrules. Putting it in the standard implies that it can be trusted.

File: standard.tex (hunk 16 above)
Comment (amihaila) [reply standard.tex:145]: >Development and documentation for all electrical projects should be tracked in git and hosted on GitHub (under the team organization, waterloo-rocketry)

(see line above)

File: standard.tex (hunk 17 above)
Comment (amihaila) [reply standard.tex:193]: >When presenting a new revision, indicate what has been changed since the last review (a commit message is enough).

The list of things that need fixing is part of the review. I do not think that the review itself should be tracked in git.

File: standard.tex (hunk 15 above)
Comment (amihaila) [reply standard.tex]: Added a comment in the design review section.

File: standard.tex (hunk 18 above)
Comment (amihaila) [reply standard.tex]: Fixed.

File: standard.tex (hunk 10 above)
Comment (amihaila) [reply standard.tex:78]: added a note

File: standard.tex (hunk 9 above)
Comment (amihaila) [reply standard.tex:71]: Ideally, yes. I do not know if that's feasible when the rocket is tall enough that you can't reach most of it. Anything remote defeats the purpose.

File: standard.tex (hunk 12 above)
Comment (amihaila) [reply standard.tex]: What are the exact changes you want made?

Comment (akmorrison) [reply standard.tex]: Check that pours appear as you expect them to, the solder mask file is sane, and the copper layers don’t have any weird shorts

File: standard.tex (hunk 9 above)
Comment (akmorrison) [reply standard.tex:71]: Even magswitches aren’t terrible. Bringing the rocket down with armed deployment charges is something we used to mock other teams about, we should have some hard rules against it

File: standard.tex (hunk 13 above)
Comment (amihaila) [reply standard.tex]: fixed

File: standard.tex (hunk 12 above)
Comment (amihaila) [reply standard.tex]: fixed

File: standard.tex (hunk 14 above)
Comment (amihaila) [reply standard.tex:135]: The important thing is that the connector on the battery is female so that the battery does not short. The fact that the thing that plugs into it is male isn't the key part. 

If we specify male connectors and the connector on the battery happens to also be male for whatever reason, then this point falls apart. Either we express a preference for female terminated batteries, or we end up with "use the connector that mates with the thing you're trying to connect to" which does not deserve to be mentioned here at all.

File: standard.tex (hunk 5 above)
Comment (amihaila) [reply standard.tex:148]: @jacobdeery thoughts on this?

File: standard.tex [hunk 19]
```diff
@@ -26,67 +26,77 @@
...
-\item If you're using an RC lithium polymer battery, it should be fused (those fuckers can sink 50 amps).
+\item All wires used should be rated for the current going through them.
+\item Strain relief:
+    \begin{itemize}
+    \item Wires that get handled, moved, or torqued at any point should be strain relieved on both sides of that point.
+    \item Wires should be relieved to the point where they can survive very violent shaking (especially if the system will be going in the rocket).
+    \item Avoid tight wires between two points.
+    \end{itemize}
+\item RC lithium polymer batteries should be fused (they can sink up to 50 A).
 \item When designing a box, circuit layout, or other assembly, keep in mind that wires and electrical components do take up space. Leave clearance in the design stage; this problem should be solved before fabrication/assembly.
```
Comment (ewdyang) [inline standard.tex:81]: Would it be worthwhile to include a point about placing hotswappable components in relatively accessible locations if possible?

File: standard.tex [hunk 20]
```diff
@@ -26,67 +26,77 @@
...
+\item If you're making a crimped connection, use a proper crimp tool if at all possible. Pliers are not an acceptable alternative (see ESRA safety critical wiring guidelines). We have proper crimp tools for: generic butt splice and ring/spade connectors, Dupont connectors, and RJ45 connectors.
+\item Use ferrules for stranded wire that goes into a screw terminal.
+\item Crimped connections should be done on stranded wire only.
+\item Wire should only be soldered to connectors that are specifically made for it (eg. connectors with solder cups). Stranded wire should be used in the vast majority of cases since solid core wire becomes brittle and prone to breakage (exception: wire soldered onto protoboard should be solid core).
+\item All solder joints should be strain relieved with zipties or P clamps.
+\item If you screw up a soldering job, redo the job. When soldering surface mount components, inspect solder joints with a small magnifying glass.
 \item All connections must be tug proof.
+\item All exposed wire should be heatshrunk; all heatshrink should be clear (IREC regulation).
```
Comment (ewdyang) [inline standard.tex]: The amount of exposed wire, even if heatshrunk, should be minimized and insulation should go up as close to a connector as possible (based on feedback from one of the judges during the poster session).

Comment (amihaila) [reply standard.tex]: fixed.

File: standard.tex (hunk 19 above)
Comment (amihaila) [reply standard.tex:81]: fixed.

**Review by akmorrison — APPROVED** [review]

**Review by pauladam316 — APPROVED** [review]

File: standard.tex (hunk 5 above)
Comment (jacobdeery) [reply standard.tex:148]: Design Documentation

A document outlining the major decisions made for your project. There are two purposes for this: first, so that future generations maintaining your project know the reasoning behind your decisions; second, because writeup on all systems is required in the IREC report. The following information is needed for the report, but feel free to add any other documentation that you think will be helpful.

- Description of the system's purpose
- Relevant design constraints and criteria
- Block diagrams (and other diagrams, if you'd like)
- Description of the system's functionality
- Description of the relevant design decisions/tradeoffs involved in developing the system (dev logs are great)
- Descriptions of any analysis/testing performed on the system

All other required documentation (checklists, failure modes/troubleshooting) is described elsewhere.

**Review by jacobdeery — APPROVED** [review]

**Comment by amihaila** [comment]
This got merged in ages ago
